Xamarin.Forms.Platform.iOS/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,267 @@
+
+				return true;
+			}
+
+			if (view is UIWebView)
+			{
+				return true;
+			}
+
+			if (view is UIScrollView scrollView)
+			{
+				if (scrollView.ContentOffset.Y < 0)
+					return true;
+
+				if (refreshing)
+					scrollView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY - _refreshControlHeight), true);
+				else
+					scrollView.SetContentOffset(new CoreGraphics.CGPoint(0, _origininalY), true);
+
+				return true;
+			}
+
+			if (view.Subviews == null)
+				return false;
+
+			for (int i = 0; i < view.Subviews.Length; i++)
+			{
+				var control = view.Subviews[i];
+				if (TryOffsetRefresh(control, refreshing))
+					return true;
+			}
+
+			return false;
+		}
+
+		bool TryInsertRefresh(UIView view, int index = 0)
+		{
+			_refreshControlParent = view;
+
+			if (view is UITableView tableView)
+			{
+				if (CanUseRefreshControlProperty())
+					tableView.RefreshControl = _refreshControl;
+				else
+					tableView.InsertSubview(_refreshControl, index);
+
+				_origininalY = tableView.ContentOffset.Y;
+				_refreshControlHeight = _refreshControl.Frame.Size.Height;
+
+				return true;
+			}
+
+			if (view is UICollectionView collectionView)
+			{
+				if (CanUseRefreshControlProperty())
+					collectionView.RefreshControl = _refreshControl;
+				else
+					collectionView.InsertSubview(_refreshControl, index);
+
+				_origininalY = collectionView.ContentOffset.Y;
+				_refreshControlHeight = _refreshControl.Frame.Size.Height;
+
+				return true;
+			}
+
+			if (view is UIWebView webView)
+			{
+				webView.ScrollView.InsertSubview(_refreshControl, index);
+				return true;
+			}
+
+			if (view is UIScrollView scrollView)
+			{
+				if (CanUseRefreshControlProperty())
+					scrollView.RefreshControl = _refreshControl;
+				else
+					scrollView.InsertSubview(_refreshControl, index);
+
+				scrollView.AlwaysBounceVertical = true;
+
+				_origininalY = scrollView.ContentOffset.Y;
+				_refreshControlHeight = _refreshControl.Frame.Size.Height;
+
+				return true;
+			}
+
+			if (view.Subviews == null)
+				return false;
+
+			for (int i = 0; i < view.Subviews.Length; i++)
+			{
+				var control = view.Subviews[i];
+				if (TryInsertRefresh(control, i))
+					return true;
+			}
+
+			return false;
+		}
+
+		void UpdateColors()
+		{
+			if (Element == null || _refreshControl == null)
+				return;
+
+			if (Element.RefreshColor != Color.Default)
+				_refreshControl.TintColor = Element.RefreshColor.ToUIColor();
+
+			SetBackgroundColor(Element.BackgroundColor);
+		}
+
+		void UpdateIsRefreshing()
+		{
+			IsRefreshing = Element.IsRefreshing;
+		}
+
+		void UpdateIsEnabled()
+		{
+			if (Element.IsEnabled)
+				TryInsertRefresh(_refreshControlParent);
+			else
+			{
+				if (_refreshControl.Superview != null)
+					_refreshControl.RemoveFromSuperview();
+			}
+		}
+
+		bool CanUseRefreshControlProperty()
+		{
+			return Forms.IsiOS10OrNewer && !_usingLargeTitles;
```

Comment:
On [line 45](https://github.com/xamarin/Xamarin.Forms/pull/7214/files#diff-b2706bb9ae1c67c18d702987c43c1ed6R45) you're checking against iOS11 and here it's 10. Shouldn't this be the same?